frontend/packages/console-app/src/components/nav/utils.ts

```diff
@@ -18,25 +18,37 @@ const itemDependsOnItem = (s1: NavExtension, s2: NavExtension): boolean => {
 const isPositioned = (item: NavExtension, allItems: NavExtension[]): boolean =>
   !!allItems.find((i) => itemDependsOnItem(item, i));
 
-const findIndexForItem = (item: NavExtension, currentItems: NavExtension[]): number => {
-  const { insertBefore, insertAfter } = item.properties;
-  let index = -1;
-  const before = toArray(insertBefore);
-  const after = toArray(insertAfter);
-  let count = 0;
-  while (count < before.length && index < 0) {
-    index = currentItems.findIndex((i) => i.properties.id === before[count]);
-    count++;
-  }
-  count = 0;
-  while (count < after.length && index < 0) {
-    index = currentItems.findIndex((i) => i.properties.id === after[count]);
-    if (index >= 0) {
-      index += 1;
+const findInsertBeforeIndex = (item: NavExtension, allItems: NavExtension[]): number => {
+  return toArray(item.properties.insertBefore).reduce((index, currentItem) => {
+    // take only the first index found
+    if (index < 0) {
+      const newIndex = allItems.findIndex((i) => i.properties.id === currentItem);
+      if (newIndex !== -1) {
+        return newIndex;
+      }
```

Comment:
Technically we don't need the extra `if` here. We can just return the result of findIndex.
```suggestion
      return allItems.findIndex((i) => i.properties.id === currentItem);
```